mlflow/server/js/src/experiment-tracking/pages/experiment-evaluation-datasets/ExperimentEvaluationDatasetsPage.tsx

```diff
@@ -19,6 +20,7 @@ const ExperimentEvaluationDatasetsPageImpl = () => {
   const [datasetListHidden, setDatasetListHidden] = useState(false);
   const [selectedDataset, setSelectedDataset] = useState<EvaluationDataset | undefined>(undefined);
   const [isDatasetsLoading, setIsDatasetsLoading] = useState(false);
+  const [selectedDatasetId, setSelectedDatasetId] = useSelectedDatasetBySearchParam();
```

Comment:
rather than using the hook here and passing it down to `ExperimentEvaluationDatasetsListTable`, can we just use it in `ExperimentEvaluationDatasetsListTable` directly?

Reply:
oh i think i see what the intention was. can we define a callback that combines `setSelectedDataset` and `setSelectedDatasetId` and pass that in instead (similar to what we did in https://github.com/mlflow/mlflow/pull/19142/files#diff-e2232542c52cc97ec1dd208f20ebfc9ded43289282fa3c8f378b8aa1a318445bR109:

```suggestion
  const [selectedDatasetId, setSelectedDatasetId] = useSelectedDatasetBySearchParam();
  
  const onSelectDataset = useCallback((dataset) => {
    setSelectedDataset(dataset);
    setSelectedDatasetId(dataset.dataset_id);
  });
```